worldengine/draw.py

```diff
@@ -458,9 +462,9 @@ def draw_satellite(world, target):
                 # The amplified difference is now translated into the rgb of the tile.
                 # This adds light to tiles higher that the previous average, and shadow
                 # to tiles lower than the previous average
-                r += adjusted_difference
-                g += adjusted_difference
-                b += adjusted_difference
+                r = max(min(adjusted_difference + r, 255), 0)  # NOTE: This could cause underflows and overflows
```

Comment:
I think that function might end up as never being used by anybody else because it is so small. Maybe we can make use of [a numpy-function](http://docs.scipy.org/doc/numpy/reference/generated/numpy.clip.html) or something similar?